duckduckgo/BrowserServicesKit
# Better detection of AMP pages

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1202198075905303/1202190330709645/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2284
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2001
What kind of version bump will this require?: Patch

**Optional**:

Tech Design URL:
CC:

**Description**:
This PR modifies the AMP feature to check for on page elements to determine if a link is an AMP link preventing false positives. It also implements the `deepExtractionEnabled` key allowing us to remotely disable deep extraction if needed.



**Steps to test this PR**:
1. Use [this config](https://www.jsonblob.com/1187494370181242880)
1. Visit freecodecamp.org and click "Get Started". It should take you to a login/signup page
2. Visit theprisonerwinecompany.com. Add something to your cart and click checkout. It should take you to the checkout page.
3. Visit golf.com, click an article, it should load without looping.
4. Visit https://privacy-test-pages.site/privacy-protections/amp/ and test the "First Party Cloaked" links. They should take you to the expected url.



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
